framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -51,29 +79,60 @@ public Void visitAnnotation(AnnotationTree node, Void p) {
         }
 
         Element element = TreeInfo.symbol((JCTree) node.getAnnotationType());
-        if (!(element.toString().equals(ArrayLen.class.getName())
+
+        if (element.toString().equals(IntRange.class.getName())) {
+            if (args.size() == 2
```

Comment:
Or, if something cannot occur, you can also leave out the test entirely, especially if the code is guaranteed to crash quickly (as it is in this case).

If the code would behave strangely and be difficult to debug, then an assert statement or an exception thrown can help with debugging.